src/main/java/de/tum/in/www1/artemis/service/AuthorizationCheckService.java

```diff
@@ -681,6 +682,10 @@ public void checkIsAdminElseThrow(@Nullable User user) {
      */
     @CheckReturnValue
     public boolean isUserAllowedToGetResult(Exercise exercise, StudentParticipation participation, Result result) {
+        if (isAtLeastInstructorForExercise(exercise) && participation.isTestRun()) {
+            return true;
+        }
+
         return isAtLeastStudentForExercise(exercise) && (isOwnerOfParticipation(participation) || isAtLeastInstructorForExercise(exercise))
                 && ExerciseDateService.isAfterAssessmentDueDate(exercise) && result.getAssessor() != null && result.getCompletionDate() != null;
```

Comment:
I think this can be written in a more general form (since a Instructor has access to any result):

```java
         if (isAtLeastInstructorForExercise(exercise)) {
            return true;
        }

        return isAtLeastStudentForExercise(exercise) && isOwnerOfParticipation(participation)
                && ExerciseDateService.isAfterAssessmentDueDate(exercise) && result.getAssessor() != null && result.getCompletionDate() != null;
```

Reply:
I am not sure if this is a better approach as instructors might access results before the end of the due date and the system could get into unexpected states.

What is the benefit of being less restrictive here?